packages/blockchain/test/4444Optimistic.spec.ts

```diff
@@ -0,0 +1,308 @@
+import { createBlock } from '@ethereumjs/block'
+import { Mainnet, createCustomCommon } from '@ethereumjs/common'
+import { equalsBytes } from '@ethereumjs/util'
+import { assert, describe, it } from 'vitest'
+
+import mergeGenesisParams from '../../client/test/testdata/common/mergeTestnet.json'
+import { createBlockchain } from '../src/index.js'
+
+describe('[Blockchain]: 4444/optimistic spec', () => {
+  const common = createCustomCommon(mergeGenesisParams, Mainnet, { name: 'post-merge' })
+  const genesisBlock = createBlock({ header: { extraData: new Uint8Array(97) } }, { common })
+
+  const block1 = createBlock(
+    { header: { number: 1, parentHash: genesisBlock.hash(), difficulty: 100 } },
+    { common, setHardfork: true },
+  )
+  const block2 = createBlock(
+    { header: { number: 2, parentHash: block1.hash(), difficulty: 100 } },
+    { common, setHardfork: true },
+  )
+  const block3PoS = createBlock(
+    { header: { number: 3, parentHash: block2.hash() } },
+    { common, setHardfork: true },
+  )
+  const block4 = createBlock(
+    { header: { number: 4, parentHash: block3PoS.hash() } },
+    { common, setHardfork: true },
+  )
+  const block5 = createBlock(
+    { header: { number: 5, parentHash: block4.hash() } },
+    { common, setHardfork: true },
+  )
+  const block6 = createBlock(
+    { header: { number: 6, parentHash: block5.hash() } },
+    { common, setHardfork: true },
+  )
+  const block7 = createBlock(
+    { header: { number: 7, parentHash: block6.hash() } },
+    { common, setHardfork: true },
+  )
+  const block8 = createBlock(
+    { header: { number: 8, parentHash: block7.hash() } },
+    { common, setHardfork: true },
+  )
+
+  const block51 = createBlock(
+    { header: { number: 5, parentHash: block4.hash(), gasLimit: 999 } },
+    { common, setHardfork: true },
+  )
+  const block61 = createBlock(
+    { header: { number: 6, parentHash: block51.hash() } },
+    { common, setHardfork: true },
+  )
+  const block71 = createBlock(
+    { header: { number: 7, parentHash: block61.hash() } },
+    { common, setHardfork: true },
+  )
+  const block81 = createBlock(
+    { header: { number: 8, parentHash: block71.hash() } },
+    { common, setHardfork: true },
+  )
+
+  it('should allow optimistic non canonical insertion unless specified explictly', async () => {
+    let dbBlock
+    const blockchain = await createBlockchain({ genesisBlock, common, validateBlocks: false })
+
+    // randomly insert the blocks with gaps will be inserted as non canonical
+    for (const block of [block4, block3PoS, block7, block2]) {
+      // without explicit flag their insertion will fail as parent not present and canonicality
+      // can't be established
+      await blockchain.putBlock(block).catch((_e) => {
+        undefined
+      })
+      dbBlock = await blockchain.getBlock(block.hash()).catch((_e) => undefined)
```

Comment:
true but since the condition below checks for it to be not undefined, the reason doesnt matter  for the purposes of the test and will need to be debugged